packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -149,16 +147,16 @@ class Icon extends StatelessWidget {
   /// This property has no effect if the [icon]'s [IconData.matchTextDirection]
   /// field is false, but for consistency a text direction value must always be
   /// specified, either directly using this property or using [Directionality].
-  final TextDirection textDirection;
+  final TextDirection? textDirection;
 
   @override
   Widget build(BuildContext context) {
     assert(this.textDirection != null || debugCheckHasDirectionality(context));
-    final TextDirection textDirection = this.textDirection ?? Directionality.of(context);
+    final TextDirection? textDirection = this.textDirection ?? Directionality.of(context);
```

Comment:
this can be non-nullable, since the assert before ensures that a directionality is available

Reply:
Done